include/deal.II/matrix_free/fe_evaluation.h

```diff
@@ -1505,22 +1604,177 @@ class FEEvaluationAccess<1,1,Number> : public FEEvaluationBase<1,1,Number>
  * real space. Currently, other finite elements cannot be treated with the
  * matrix-free concept.
  *
- * This class has five template arguments:
+ * <h3>Handling multi-component systems</h3>
  *
- * @param dim Dimension in which this class is to be used
+ * FEEvaluation also allows for treating vector-valued problems through a
+ * template parameter on the number of components:
+ *
+ * @code
+ * FEEvaluation<dim,fe_degree,n_q_points_1d,n_components> phi(matrix_free);
+ * @endcode
+ *
+ * If used this way, the components can be gathered from several components of
+ * an @p std::vector<VectorType> through the call
+ *
+ * @code
+ * phi.read_dof_values(src, 0);
+ * @endcode
+ *
+ * where the 0 means that the vectors starting from the zeroth vector in the
+ * @p std::vector should be used, <code>src[0], src[1], ...,
+ * src[n_components-1]</code>.
  *
- * @param fe_degree Degree of the tensor product finite element with
+ * An alternative way for reading multi-component systems is possible if the
+ * DoFHandler underlying the MatrixFree data is based on an FESystem of @p
+ * n_components entries. In that case, a single vector is provided for the
+ * read_dof_values() and distribute_local_to_global() calls.
+ *
+ * An important property of FEEvaluation in multi-component systems is the
+ * layout of multiple components in the get_value(), get_gradient(), or
+ * get_dof_value() calls. In this case, instead of a scalar return field
+ * VectorizedArray@<double@> a tensor is returned,
+ *
+ * @code
+ * get_value -> Tensor<1,n_components,VectorizedArray<double> >
+ * get_gradient -> Tensor<1,n_components,Tensor<1,dim,VectorizedArray<double> > >
+ * @endcode
+ *
+ * In a similar vein, the submit_value() and submit_gradient() calls take
+ * tensors of values. Note that there exist specializations for @p
+ * n_components=1 and @p n_components=dim, which are provided through the base
+ * class FEEvaluationAccess. In the scalar case, these provide the scalar
```

Comment:
Yes, it is a detail of implementation. Nonetheless, if one wants to look at what type the `get_value()` methods return (or the fact that is a `get_divergence` call for the specialization `n_components==dim`), one needs to be aware of that class. Since I'm one of two authors of that part and know where things are, I'd be happy to let you or @davydden decide whether it is important to mention that here.